api/app/controllers/webhook.controller.ts

```diff
@@ -235,18 +235,36 @@ async function processCommit(
 
     let transcriptData: any;
 
+    async function cacheTranscript(transcriptData: any) {
+      const redisTransaction = redis.multi();
+      redisTransaction.sadd("cachedTranscripts", transcriptData.id);
+      redisTransaction.set(
+        `transcript:${transcriptData.id}`,
+        JSON.stringify(transcriptData),
+        "EX",
+        CACHE_EXPIRATION
+      );
+      const totalItems = await Transcript.count();
+      const totalPages = Math.ceil(totalItems / DB_QUERY_LIMIT);
+      for (let page = 1; page <= totalPages; page++) {
+        redisTransaction.del(`transcripts:page:${page}`);
+      }
```

Comment:
same here about the re-usable function

Reply:
If you see the webhook controller, I already scoped the Redis flow to its own function. I can extend it to other controllers then.